### ISSUE TITLE
Determine compatibility with DigiCert's ACME endpoint

### ISSUE THREAD

**Issue by mholt** [go-acme/lego#919]
DigiCert now has ACME support: https://docs.digicert.com/certificate-tools/acme-user-guide/

It'd be worth checking to make sure lego's ACME client can successfully interact with it. 

In particular, custom order fields are necessary: https://docs.digicert.com/certificate-tools/acme-user-guide/acme-beta-known-issues/

**Comment by ozkar99** [comment]
DISCLAIMER: **I am using v2.7.2** will update to the latest ASAP and re-test.

Hey, y'all I am using this for DigiCert acme integration and **can confirm it doesn't work**.

So far I've found that:

a) Digicert ACME implementation doesn't play well with custom `ObtainForCSR`, you get a malformed request error but not much of a description or explanation on the error. (for context the same exact code works on let's encrypt without issues).

b) If you use `Obtain` with no CSR the order finalizes successfully BUT the last step of downloading the certificate times out (even with a 5-minute timeout).

I am waiting on Digicert support to get back to some of my questions to help me diagnose both errors.

I'll be updating this ticket as I get more info.

Cheers!

**Comment by mholt** [comment]
@ozkar99 It's likely the custom CSR you created is incompatible with their endpoint, whereas the default lego creates is compatible.

As for the timeout downloading the cert, that's likely a bug they'll have to fix...

**Comment by ozkar99** [comment]
> @ozkar99 It's likely the custom CSR you created is incompatible with their endpoint, whereas the default lego creates is compatible.

Yeah, this looks to be the case. Overall I am ok with this, as they use all of the CSR fields from the organization registry account (it might even be the root cause).

> As for the timeout downloading the cert, that's likely a bug they'll have to fix...

Yeah most likely, although `certbot` fully issues certificates and downloads then without any issues (and it doesn't take more than 1-2 seconds after challenges have been solved so it's doesn't seem to be an issue with the cert not being available). I am waiting on them to get back on that issue.

I'll update as soon as I have more info on that.

**Comment by ozkar99** [comment]
Hey folks, after working around the issue where Digicert will fail the previously mentioned custom CSR (they only allow `CommonName` and the `SAN` names on the CSR) 

I can confirm that this library successfully handles the Digicert ACME implementation 🥳 .

Thanks, Matt and Ludovic for your hard work around ACME+go!

**Comment by mholt** [comment]
@ozkar99 Good to know, thanks. Can you clarify though? What was on your CSR that made it fail? And did DigiCert actually fix their bug or did you just generate a different CSR?

**Comment by ozkar99** [comment]
Oh, whenever I filled anything outside of `CommonName` and `SAN` it would fail.

So:

- Organization
- Country
- State
- Locality

They didn't really fix their bug, but I adjusted the CSR I generated by leaving out those fields, to work around it.

For the record, Digicert does fill those fields on the end certificate with the information they have on the account you use to access their ACME directory.

However seeing as both `Obtain` and `ObtainForCSR` work fine, I don't think this is a `lego` issue but rather a Digicert one.
